jenkins_job_wrecker/modules/publishers.py

```diff
@@ -78,30 +78,104 @@ def fingerprinter(top, parent):
         elif element.tag == 'presendScript':
             ext_email['presend-script'] = element.text
+        elif element.tag == 'postsendScript':
+            ext_email['postsend-script'] = element.text
         elif element.tag == 'sendTo':
             ext_email['send-to'] = element.text
         elif element.tag == 'configuredTriggers':
-            print("IGNORED configuredTriggers in email-ext")
+            trigger_prefix = "hudson.plugins.emailext.plugins.trigger."
+            trigger_dict = {
+                "AlwaysTrigger": "always",
+                "UnstableTrigger": "unstable",
+                "FirstFailureTrigger": "first-failure",
+                "FirstUnstableTrigger": "first-unstable",
+                "NotBuiltTrigger": "not-built",
+                "AbortedTrigger": "aborted",
+                "RegressionTrigger": "regression",
+                "FailureTrigger": "failure",
+                "SecondFailureTrigger": "second-failure",
+                "ImprovementTrigger": "improvement",
+                "StillFailingTrigger": "still-failing",
+                "SuccessTrigger": "success",
+                "FixedTrigger": "fixed",
+                "FixedUnhealthyTrigger": "fixed-unhealthy",
+                "StillUnstableTrigger": "still-unstable",
+                "PreBuildTrigger": "pre-build"
+            }
+            trigger_result = {}  # To be able to print all on the output yaml
+            for key in trigger_dict.keys():
+                trigger_result[key] = False
+            # JJB implementation only uses the default content-type/subject/message
+            # and uses those in the trigger elements. Does not take information
+            # from trigger element's content. However, sendTo must be taken from
+            # one of the triggers, in this case, first trigger will be used in order
+            # to retrieve sendTo information.
+            is_send_to_received = False
+            for trigger in element:
+                for email in trigger:
+                    if not is_send_to_received:
+                        if email.tag == "email":
+                            send_to = []
+                            for email_child in email:
+                                if email_child.tag == "sendToDevelopers":
+                                    send_to.append("developers")
+                                elif email_child.tag == "sendToRequester":
+                                    send_to.append("requester")
+                                elif email_child.tag == "includeCulprits":
+                                    send_to.append("culprits")
+                                elif email_child.tag == "sendToRecipientList":
+                                    send_to.append("recipients")
+                                elif email_child.tag == "recipientProviders":
+                                    for provider in email_child:
+                                        if provider.tag == \
```

Comment:
What about `else:`?

I found this in one of STB Tests: `hudson.plugins.emailext.plugins.recipients.ListRecipientProvider`. Is it covered somehow?

Reply:
That is not covered in JJB, JJB needs to be updated for that

Reply:
I think at this moment we should focus on our use cases. If we need to update JJB to cover STB tests then we need to do it.